nixos/tests/kafka/mirrormaker.nix

```diff
@@ -0,0 +1,242 @@
+import ../make-test-python.nix (
+  { lib, pkgs, ... }:
+
+  let
+    inherit (lib) mkMerge;
+
+    clusterAId = pkgs.runCommand "kafka-cluster-a-id" { buildInputs = [ pkgs.apacheKafka ]; } ''
+      kafka-storage.sh random-uuid > $out
+    '';
+
+    clusterBId = pkgs.runCommand "kafka-cluster-b-id" { buildInputs = [ pkgs.apacheKafka ]; } ''
+      kafka-storage.sh random-uuid > $out
+    '';
+
+    kafkaConfig = {
+      networking.firewall.allowedTCPPorts = [
+        9092
+        9093
+      ];
+
+      virtualisation.diskSize = 1024;
+      virtualisation.memorySize = 1024 * 2;
+
+      environment.systemPackages = [ pkgs.apacheKafka ];
+
+      services.apache-kafka = {
+        enable = true;
+
+        formatLogDirs = true;
+
+        settings = {
+          listeners = [
+            "PLAINTEXT://:9092"
+            "CONTROLLER://:9093"
+          ];
+          "listener.security.protocol.map" = [
+            "PLAINTEXT:PLAINTEXT"
+            "CONTROLLER:PLAINTEXT"
+          ];
+          "controller.listener.names" = [ "CONTROLLER" ];
+
+          "process.roles" = [
+            "broker"
+            "controller"
+          ];
+
+          "log.dirs" = [ "/var/lib/apache-kafka" ];
+          "num.partitions" = 1;
+          "offsets.topic.replication.factor" = 1;
+          "transaction.state.log.replication.factor" = 1;
+          "transaction.state.log.min.isr" = 1;
+        };
+      };
+
+      systemd.services.apache-kafka = {
+        after = [ "network-online.target" ];
+        requires = [ "network-online.target" ];
+        serviceConfig.StateDirectory = "apache-kafka";
+      };
+    };
+
+    extraKafkaConfig = {
+      kafkaa1 = {
+        services.apache-kafka = {
+          clusterId = builtins.readFile "${clusterAId}";
```

Comment:
It seems the f-string strick won't work here, consider hard-coding the uuids

Reply:
Put back in the hardcoded UUIDs now.